epicbox/utils.py

```diff
@@ -27,11 +27,12 @@
 
 
 def get_docker_client(base_url=None, retry_read=config.DOCKER_MAX_READ_RETRIES,
-                      retry_status_forcelist=(500,)):
+                      retry_status_forcelist=(500,), tls=False):
```

Comment:
Currently retries for unsuccessful requests to Docker engine are only applied for `http://` base url: https://github.com/StepicOrg/epicbox/blob/master/epicbox/utils.py#L35-L43. When using tls a docker client mounts an [`SSLAdapter`](https://github.com/docker/docker-py/blob/master/docker/tls.py#L108-L112) that needs to be patched/replaced as well. Otherwise the behaviour of the epicbox library may differ depending on whether tls is used or not.
In general, retries help recover occasional Docker errors, e.g., when writing files to a sandbox: https://github.com/StepicOrg/epicbox/blob/master/epicbox/sandboxes.py#L273-L276